napari/layers/labels/_tests/test_labels.py

```diff
@@ -1562,6 +1562,22 @@ def test_color_mapping_when_seed_is_changed():
     assert not np.allclose(mapped_colors1, mapped_colors2)
 
 
+def test_color_shuffling_above_num_colors():
+    """Check that the color shuffle does not result in the same collisions.
+
+    See https://github.com/napari/napari/issues/6448.
+    """
+    # period: this is the number of non-background colors in the colormap
+    period = 50 - 1
+    labels = np.arange(1, 1 + 2 * period).reshape((2, period))
+    layer = Labels(labels)
+    colors0 = layer.colormap.map(labels)
+    assert np.all(colors0[0] == colors0[1])
+    layer.new_colormap()
+    colors1 = layer.colormap.map(labels)
+    assert not np.all(colors1[0] == colors1[1])
```

Comment:
I have created a modified version of this PR, based on pointed problems with big numbers, and it looks like your workaround is nor correct.

```python
def test_color_shuffling_above_num_colors_254():
    """Check that the color shuffle does not result in the same collisions.

    See https://github.com/napari/napari/issues/6448.
    """
    # period: this is the number of non-background colors in the colormap
    period = 254
    labels = np.arange(1, 1 + 2 * period).reshape((2, period))
    layer = Labels(labels)
    layer.num_colors = period
    colors0 = layer.colormap.map(labels)
    assert np.all(colors0[0] == colors0[1])
    layer.new_colormap()
    colors1 = layer.colormap.map(labels)
    assert not np.all(colors1[0] == colors1[1])
```

ends with error:

```
        assert np.all(colors0[0] == colors0[1])
>       layer.new_colormap()

napari/layers/labels/_tests/test_labels.py:1592: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
napari/layers/labels/labels.py:1123: in new_colormap
    self.seed_rng = np.random.default_rng().integers(2**32 - 1)
napari/layers/labels/labels.py:516: in seed_rng
    self._random_colormap = shuffle_and_extend_colormap(
napari/utils/colormaps/colormap_utils.py:547: in shuffle_and_extend_colormap
    rng.choice(indices, size=n_colors - n_colors_prev)
numpy/random/_generator.pyx:850: in numpy.random._generator.Generator.choice
    ???
numpy/random/_generator.pyx:624: in numpy.random._generator.Generator.integers
    ???
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

>   ???
E   ValueError: negative dimensions are not allowed
```

Reply:
I think that if there are not enough primes, we should just shuffle values.

Reply:
Just shuffling will not close #6448 in those situations. I've pushed a fix that shuffles but shortens the colormap appropriately. (I did notice the missing coverage report yesterday but it was just as I was going to bed 😅)